OpenWeather/App/Search/SearchScreen.swift

```diff
@@ -43,16 +43,24 @@ struct SearchScreen: View {
                 LocationResultItem(name: location.name)
             }
         }
-        .searchable(text: $searchText, prompt: "Search by name or zipcode")
+        .searchable(
+            text: Binding<String>(
+                get: { viewModel.searchText },
+                set: { viewModel.searchText = $0 }
+            ),
+            prompt: "Search by name or zipcode"
+        )
         .navigationDestination(
             for: Date.self,
             destination: { date in
                 Text(date, format: .dateTime)
             }
         )
         .navigationTitle("Search")
-        .onChange(of: searchText) { newSearchText in
-            viewModel.getLocations(query: newSearchText)
+        .onReceive(
+            viewModel.$searchText.debounce(for: 0.300, scheduler: RunLoop.main)
+        ) { _ in
+            viewModel.getLocations()
```

Comment:
After we adopt the ViewModel package we can fully move this into the ViewModel.. but this is fine for now.